File: music-lessons-2-java/app/src/main/java/com/mstoyanov/musiclessons/model/WeekdayConverter.java
package com.mstoyanov.musiclessons.model;

import android.arch.persistence.room.TypeConverter;

public class WeekdayConverter {
    @TypeConverter
    public Weekday toWeekdayName(String value) {
        switch (value) {
            case "Monday":
                return Weekday.MONDAY;
            case "Tuesday":
                return Weekday.TUESDAY;
            case "Wednesday":
                return Weekday.WEDNESDAY;
            case "Thursday":
                return Weekday.THURSDAY;
            case "Friday":
                return Weekday.FRIDAY;
            case "Saturday":
                return Weekday.SATURDAY;
            case "Sunday":
                return Weekday.SUNDAY;
            default:
                return null;
        }
    }

    @TypeConverter
    public String toString(Weekday weekday) {
        return weekday.displayValue();
    }
}

File: music-lessons-2-kotlin/app/src/main/java/com/mstoyanov/musiclessons/model/WeekdayConverter.kt
package com.mstoyanov.musiclessons.model

import android.arch.persistence.room.TypeConverter

class WeekdayConverter {
    @TypeConverter
    fun toWeekdayName(value: String): Weekday? {
        return when (value) {
            "Monday" -> Weekday.MONDAY
            "Tuesday" -> Weekday.TUESDAY
            "Wednesday" -> Weekday.WEDNESDAY
            "Thursday" -> Weekday.THURSDAY
            "Friday" -> Weekday.FRIDAY
            "Saturday" -> Weekday.SATURDAY
            "Sunday" -> Weekday.SUNDAY
            else -> null
        }
    }

    @TypeConverter
    fun toString(weekday: Weekday): String {
        return weekday.displayValue()
    }
}